### ISSUE TITLE
Fix QR code to house IPFS hash, encrypted prescription, and smart contract address

### ISSUE THREAD

**Issue by pi0neerpat** [blockchainbuddha/Pharma-Sign#3]

**Comment by moisesja** [comment]
It sounds to me that each individual Rx will have its own contract instance, and truly give it a state: prescribed, filled, rejected, etc...

Let me know what are your thoughts.

**Comment by pi0neerpat** [comment]
I agree that each Rx state should be on-chain. But they can still all be in the same contract. Re-deploying a new contract each rx adds gas and complexity
